tests/MongoDB.Driver.Tests/Specifications/command-monitoring/BulkWriteTest.cs

```diff
@@ -48,32 +49,53 @@ protected override bool TrySetArgument(string name, BsonValue value)
                 case "requests":
                     _requests = ParseRequests((BsonArray)value).ToList();
                     return true;
+                case "options":
+                    _options = ParseOptions(value.AsBsonDocument);
+                    return true;
                 case "ordered":
                     _options.IsOrdered = value.ToBoolean();
                     return true;
-                case "writeConcern":
-                    _writeConcern = WriteConcern.FromBsonDocument((BsonDocument)value);
-                    return true;
             }
 
             return false;
         }
+        
+        // private methods
+        private BulkWriteOptions ParseOptions(BsonDocument value)
+        {
+            var options = new BulkWriteOptions();
+
+            foreach (var option in value.Elements)
+            {
+                switch (option.Name)
+                {
+                    case "ordered":
+                        options.IsOrdered = option.Value.ToBoolean();
+                        break;
+                    default:
+                        throw new FormatException($"Unexpected option: ${option.Name}.");
+                }
+            }
+
+            return options;
+        }
 
         private IEnumerable<WriteModel<BsonDocument>> ParseRequests(BsonArray requests)
         {
             foreach (BsonDocument request in requests)
             {
-                var element = request.GetElement(0);
-                switch (element.Name)
+                var name = request["name"].AsString;
+                var arguments = request["arguments"].AsBsonDocument;
+                switch (name)
                 {
                     case "deleteOne":
-                        yield return ParseDeleteOne((BsonDocument)element.Value);
+                        yield return ParseDeleteOne(arguments);
```

Comment:
Sounds like this `yield` has no sense at all in these lines. I understand that this is not your change, but since you're making some small changes here, consider to replace this `yield` logic on just `List<WriteModel<BsonDocument>>`